router/handler_edge_ctrl/apiSessionAdded.go

```diff
@@ -93,61 +88,61 @@ func (h *apiSessionAddedHandler) HandleReceive(msg *channel2.Message, _ channel2
 				}
 
 				h.reqChan <- reqWithState
-			} else {
-				for _, session := range req.ApiSessions {
-					h.sm.AddApiSession(session)
+			} else if h.sm.IsSyncInProgress() {
+				reqWithState := &apiSessionAddedWithState{
```

Comment:
I assume the assumption is that if this sync fails, the ApiSession created during the sync will be present in the subsequent sync.

Reply:
Yes